packages/auth/__tests__/providers/cognito/signInResumable.test.ts

```diff
@@ -0,0 +1,278 @@
+		ADD_OAUTH_LISTENER: jest.fn(() => ({})),
+	},
+	syncSessionStorage: {
+		setItem: jest.fn((key, value) => {
+			window.sessionStorage.setItem(key, value);
+		}),
+		getItem: jest.fn((key: string) => {
+			return window.sessionStorage.getItem(key);
+		}),
+		removeItem: jest.fn((key: string) => {
+			window.sessionStorage.removeItem(key);
+		}),
+	},
+}));
+
+jest.mock(
+	'../../../src/providers/cognito/utils/clients/CognitoIdentityProvider',
+);
+
+const signInStateKeys: Record<string, string> = {
+	username: 'CognitoSignInState.username',
+	challengeName: 'CognitoSignInState.challengeName',
+	signInSession: 'CognitoSignInState.signInSession',
+	expiry: 'CognitoSignInState.expiry',
+};
+
+const user1: Record<string, string> = {
+	username: 'joonchoi',
+	challengeName: 'CUSTOM_CHALLENGE',
+	signInSession: '888577-ltfgo-42d8-891d-666l858766g7',
+	expiry: '1234567',
+};
+
+describe('signInStore', () => {
+	const authConfig = {
+		Cognito: {
+			userPoolClientId: '123456-abcde-42d8-891d-666l858766g7',
+			userPoolId: 'us-west-7_ampjc',
+		},
+	};
+
+	const session = '1234234232';
+	const challengeName = 'SMS_MFA';
+	const { username } = authAPITestParams.user1;
+	const { password } = authAPITestParams.user1;
+
+	beforeEach(() => {
+		cognitoUserPoolsTokenProvider.setAuthConfig(authConfig);
+	});
+
+	beforeAll(() => {
+		setUpGetConfig(Amplify);
+	});
+
+	afterEach(() => {
+		jest.clearAllMocks();
+	});
+
+	afterAll(() => {
+		jest.restoreAllMocks();
+	});
+
+	test('LocalSignInState is empty after initialization', async () => {
+		const localSignInState = signInStore.getState();
+
+		expect(localSignInState).toEqual({
+			challengeName: undefined,
+			signInSession: undefined,
+			username: undefined,
+		});
+		cleanActiveSignInState();
+	});
+
+	test('State is set after calling setActiveSignInState', async () => {
+		setActiveSignInState(user1);
+		const localSignInState = signInStore.getState();
+
+		expect(localSignInState).toEqual(user1);
+		cleanActiveSignInState();
+	});
+
+	test('State is updated after calling SignIn', async () => {
+		const handleUserSRPAuthflowSpy = jest
+			.spyOn(signInHelpers, 'handleUserSRPAuthFlow')
+			.mockImplementationOnce(
+				async (): Promise<RespondToAuthChallengeCommandOutput> => ({
+					ChallengeName: challengeName,
+					Session: session,
+					$metadata: {},
+					ChallengeParameters: {
+						CODE_DELIVERY_DELIVERY_MEDIUM: 'SMS',
+						CODE_DELIVERY_DESTINATION: '*******9878',
+					},
+				}),
+			);
+
+		await signIn({
+			username,
+			password,
+		});
+		const newLocalSignInState = signInStore.getState();
+
+		expect(handleUserSRPAuthflowSpy).toHaveBeenCalledTimes(1);
+		expect(newLocalSignInState).toEqual({
+			challengeName,
+			signInSession: session,
+			username,
+			signInDetails: {
+				loginId: username,
+				authFlowType: 'USER_SRP_AUTH',
+			},
+		});
+		handleUserSRPAuthflowSpy.mockClear();
+	});
+
+	test('Stored session is not expired thus State should be rehydrated', () => {
```

Comment:
Nit. The wording could be improved for better clarity.

```suggestion
The stored sign-in state should be rehydrated if the sign-in session is still valid.
```